lib/fp.js

```diff
@@ -54,13 +31,25 @@ const fmap = (fn1, f) => {
 //   funcA - <Function>
 const ap = (fn, funcA) => concat(funcA, (f, callback) => fmap(fn, f)(callback));
 
-asyncChainMethods = { fmap, ap, concat };
+function AsyncFunction() {}
```

Comment:
`AsyncFunction` is not ok because we have `(async x => x).constructor` and it is also `AsyncFunction`